lib/grp.gi

```diff
@@ -4291,86 +4294,121 @@ end );
 InstallMethod( GroupWithGenerators,
     "generic method for collection",
     [ IsCollection ],
-    function( gens )
-    local G,fam,typ;
-
-    fam:=FamilyObj(gens);
-    if IsFinite(gens) then
-      if not IsBound(fam!.defaultFinitelyGeneratedGroupType) then
-        fam!.defaultFinitelyGeneratedGroupType:=
-          NewType(fam,IsGroup and IsAttributeStoringRep
-                      and HasGeneratorsOfMagmaWithInverses
-                      and IsFinitelyGeneratedGroup);
-      fi;
-      typ:=fam!.defaultFinitelyGeneratedGroupType;
-    else
-      if not IsBound(fam!.defaultGroupType) then
-        fam!.defaultGroupType:=
-          NewType(fam,IsGroup and IsAttributeStoringRep
-                      and HasGeneratorsOfMagmaWithInverses);
+function( gens )
+local G,fam,typ,id,triv;
+
+  fam:=FamilyObj(gens);
+
+  triv:=fail; # might not find out
+
+  typ:=IsGroup and IsAttributeStoringRep 
+    and HasIsEmpty and HasGeneratorsOfMagmaWithInverses;
+  if IsFinite(gens) then
+    typ:=typ and IsFinitelyGeneratedGroup;
+    if Length(gens)<=1 then
+      typ:=typ and HasIsCyclic and IsCyclic 
+          and HasIsCommutative and IsCommutative;
+    fi;
+
+    if Length(gens)>0 and CanEasilyCompareElements(gens) then
+      id:=One(gens[1]);
+      if ForAny(gens,x->x<>id) then
+        typ:=typ and HasIsTrivial and HasIsNonTrivial;
+        triv:=false;
+      else
+        typ:=typ and HasIsTrivial and HasIsNonTrivial;
+        triv:=true;
+        # basic consequence
+        typ:=typ and HasIsPerfectGroup and IsPerfectGroup;
```

Comment:
I'd rather have a `InstallTrueMethod( IsPerfectGroup, IsGroup and IsTrivial );`, this will cover the above (with zero overhead).